pkg/operator/controller/dns/controller.go

```diff
@@ -480,18 +478,21 @@ func conditionChanged(a, b iov1.DNSZoneCondition) bool {
 
 // migrateRecordStatusConditions purges deprecated fields from DNS Record
 // status. Removes "Failed" (DNSRecordFailedConditionType) and replaces with
-// "Published" (DNSRecordPublishedConditionType). Returns updated condition
-// list.
+// "Published" (DNSRecordPublishedConditionType). If the status needs to be
+// updated, this function performs the update.  The function returns a Boolean
+// value indicating whether it has performed an update along with the error
+// value from any update it has performed.
 //
 // This function is at least required for 1 minor release and can be removed after
 // Openshift 4.13 is released.
 func (r *reconciler) migrateRecordStatusConditions(record *iov1.DNSRecord) (bool, error) {
 	var updateConditions bool
 	for i := range record.Status.Zones {
-		var updateZone bool
-		updateZone, record.Status.Zones[i].Conditions = migrateRecordStatusCondition(record.Status.Zones[i].Conditions)
+		zone := record.Status.DeepCopy().Zones[i]
+		updateZone, cond := migrateRecordStatusCondition(zone.Conditions)
 		if updateZone {
 			updateConditions = true
+			record.Status.Zones[i].Conditions = mergeConditions(zone.Conditions, cond)
```

Comment:
Ah, was the logic setting the last transition time incorrectly before this change?

Reply:
It wasn't merging the condition types resulting in multiple entries of the same condition type.

Reply:
There was an edge case within migration code, I've added an additional test made the required changes in migration function.